### PR TITLE
Add InnerLoop Configuration Profiles for Faster Prototyping

### PR DESCRIPTION
**Pull request by JimBobSquarePants**
### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/SixLabors/ImageSharp/pulls) open
- [x] I have verified that I am following the existing coding patterns and practice as demonstrated in the repository. These follow strict Stylecop rules :cop:.
- [x] I have provided test coverage for my change (where applicable)

### Description
Adds two new solution configuration profiles that target on a single target framework `netcoreapp3.1` in order to allow faster development prototyping due to reduce build times. See #1603
- Debug-InnerLoop
- Release-InnerLoop

In addition MinVer and SourceLink have been configured upstream via our Shared Infrastructure submodule to only be trigger when running in CI.

I've also updated our editorconfig profile to work better with VS 16.8+ #1600. There's a lot more analyzers running in VS now so we get more warnings but nothing critical.

NOTE: Visual Studio is a little flaky when you switch between the profiles (fixed by a restart) so I suggest picking a profile and sticking with it for the most part.

### CODE REVIEW

**Comment by antonfirsov** [comment]
Want to try this out with both VS and Rider this week, and come back.

File: src/ImageSharp/ImageSharp.csproj [hunk 1]
```diff
@@ -20,6 +21,11 @@
         <TargetFrameworks>net5.0;netcoreapp3.1;netcoreapp2.1;netstandard2.1;netstandard2.0;netstandard1.3;net472</TargetFrameworks>
       </PropertyGroup>
     </When>
+    <When Condition="$(Configuration.EndsWith('InnerLoop')) == true">
+      <PropertyGroup>
+        <TargetFrameworks>netcoreapp3.1</TargetFrameworks>
```
Comment (antonfirsov) [inline src/ImageSharp/ImageSharp.csproj:26]: How about this?
https://github.com/adamralph/minver#can-i-disable-minver

```suggestion
        <TargetFrameworks>netcoreapp3.1</TargetFrameworks>
        <MinVerSkip>true</MinVerSkip>
```

**Review by antonfirsov — APPROVED** [review]
Seems to work nicely for me, great change, thanks!

There is an easy way to disable MinVer without touching shared infra.

File: src/ImageSharp/ImageSharp.csproj (hunk 1 above)
Comment (JimBobSquarePants) [reply src/ImageSharp/ImageSharp.csproj:26]: Didn't know that was possible! I'll leave the changes in the submodule for now though as I don't really see a use for MinVer outside of CI.

**Pull request merged**